src/rules/spaceWithinParensRule.ts

```diff
@@ -0,0 +1,167 @@
+    size: number;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "space-within-parens",
+        description: "Enforces spaces within parentheses or disallow them.",
+        hasFix: true,
+        optionsDescription: Lint.Utils.dedent`
+            You may enforce the amount of whitespace within parentheses.
+        `,
+        options: {
+            type: "array",
+            items: [
+                {
+                    type: "number",
+                    enum: [OPTION_SPACE_SIZE_0, OPTION_SPACE_SIZE_1],
+                },
+            ],
+            minLength: 0,
+            maxLength: 4,
+        },
+        type: "style",
+        typescriptOnly: false,
+    };
+    /* tslint:enable:object-literal-sort-keys */
+
+    public static FAILURE_NO_SPACE = "No space within parentheses";
+    public static FAILURE_NEEDS_SPACE = "Needs space within parentheses";
+    public static FAILURE_NO_EXTRA_SPACE(count: number): string {
+        return `No more than ${count} space within parentheses allowed`;
+    }
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new SpaceWhitinParensWalker(sourceFile, this.ruleName, parseOptions(this.ruleArguments[0])));
+    }
+}
+
+function parseOptions(whitespaceSize: number): Options {
+    let size = 0;
+    if (typeof whitespaceSize === 'number') {
+        size = whitespaceSize;
+    } else {
+        const parsedSize = parseInt(whitespaceSize, 10);
+        if (!isNaN(parsedSize)) {
+            size = parsedSize;
+        }
+    }
+    return {
+        size,
+    };
+}
+
+class SpaceWhitinParensWalker extends Lint.AbstractWalker<Options> {
+    private openingParenRegex = /^([^\(]*\()( *)[^\n\r]*$/m;
+    private closingParenRegex = /([^\s])(\s*)\)$/;
+
+    public walk(sourceFile: ts.SourceFile) {
+        const cb = (node: ts.Node): void => {
+            this.checkNode(node);
+            ts.forEachChild(node, cb);
+        };
+        return ts.forEachChild(sourceFile, cb);
+    }
+
+    private getOpenParenToken(node: ts.Node): ts.Node | undefined {
+        return Lint.childOfKind(node, ts.SyntaxKind.OpenParenToken);
+    }
+
+    private getCloseParenToken(node: ts.Node): ts.Node | undefined {
+        return Lint.childOfKind(node, ts.SyntaxKind.CloseParenToken);
+    }
+
+    private checkNode(node: ts.Node): void {
+        let child: ts.Node | undefined;
+        if ((child = this.getOpenParenToken(node)) !== undefined) {
+            this.checkOpenParenToken(child);
+        }
+        if ((child = this.getCloseParenToken(node)) !== undefined) {
+            this.checkCloseParenToken(child);
+        }
+    }
+
+    private checkOpenParenToken(node: ts.Node) {
+        const code = this.sourceFile.text.substr(node.getFullStart());
+        const matchResult = this.openingParenRegex.exec(code);
```

Comment:
Coming from your last comment on how to find out we are facing this case `console.log()`:
We are inside of a loop (Note: I updated the code based on `tsutils`):
```ts

    private checkNode(node: ts.Node): void {
        forEachToken(node, (token: ts.Node) => {
            if (token.kind === ts.SyntaxKind.OpenParenToken) {
                this.checkOpenParenToken(token);
            } else if (token.kind === ts.SyntaxKind.CloseParenToken) {
                this.checkCloseParenToken(token);
            }
        });
    }
```
how I'm supposed to know if I already passed the `checkOpenParenToken` to cancel the next call which is `checkCloseParentToken` of this case?